java/arcs/core/entity/EntityBase.kt

```diff
@@ -261,14 +269,14 @@ open class EntityBase(
         schema.fields.singletons.keys
       ).map { field ->
         field to getSingletonValue(field)?.let {
-          toReferencable(it, getSingletonType(field))
+          toReferencable(it, getSingletonType(field))!!
```

Comment:
Prefer requireNotNull with an error message in production code. `!!` is good for tests, where the error message isn't so important.

Reply:
Replaced the others and allowed the nullablility in this case (i.e. removed the `!!` operator) as null values are actually allowed as field values.

Reply:
Not done? I still see `!!` here.